FEAT: MassDownloader rollout

hemmelig opened the pull request:
<!--
Thank your for contributing to QuakeMigrate!
Please fill out the sections below.
-->

## What is the purpose of this Pull Request? 
Roll out the MassDownloader to move from bundling data for the included examples to downloading it from IRIS/EarthScope at runtime. Also move from Client -> MassDownloader for the `Volcanotectonic_Iceland` example, to take advantage of the efficiency gains it provides.

Secondarily, address #158 (and allow the `Icequake_Rutford` example to still produce plots with the IRIS-provided data, which is labelled Z->1, N->2 and E->3 for the three components) by adding the capability to determine component lists for each phase from the user-supplied `onset.channel_maps` dict.

Not yet done:
 - [x] Update notebooks for the two Icequake examples (always forget these..)

### Relevant Issues
#158 

## Test cases
Please detail any tests that you have used to ensure your changes do not introduce bugs.

- [x] All examples run without any new warnings
- [x] test_benchmarks.py reports all example tests pass
    - `Icequake_Iceland` raw cut waveforms have been updated, as detailed in the commit.

### System details
Ubuntu 22.04 / Python 3.12.5 / AMD Epyc CPU

### Final checklist
- [x] Correct base branch selected?
    - `master` (if a bugfix/patch update)
    - `version/vX.X.X` if a new significant feature (discuss with developers)
- [x] All tests still pass.
~- [ ] Any new features or fixed regressions are covered by new tests.~
~- [ ] Any new or changed features are fully documented.~
- [x] Significant changes have been added to `CHANGES.md`.

Reopening of #185, in order to finish last changes.

Original PR authored by @TomWinder.

hemmelig committed:
:wrench: Update notebooks

Update example problem notebooks to use the data downloader and channel
mappings, per the changes made to the example scripts.

TomWinder committed:
:bug: Fix paths to locate outputs in notebooks

Minor fixes to paths, due to trigger tweaks affecting EventID's (both
notebooks) and to fix a mismatch in run naming ('icequake_example' -->
'example_run') for the Rutford notebook in 5d7510f

Also re-use the "data_in" variable when setting up the path for the
MassDownloader.

TomWinder committed:
:bookmark_tabs: Update CHANGES.md

Add changes from #185 and #186.

TomWinder reviewed (dismissed):
I've updated `CHANGES.md` and made some minor fixes to the notebooks. Maybe worth a final check all parts of the notebooks run through on your end, but otherwise good to merge.

hemmelig committed:
:bug: Fix filepath issue in icequake notebook

Fix filepath issue in Iceland icequake example notebook, where an old
.event file was specified before the example was improved.
